lib/logic/models/content.dart

```diff
@@ -26,6 +26,15 @@ abstract class Content with EquatableMixin {
 
   /// Enumerates all the types of content (derived from this class).
   static List<Type> enumerate() => [Song, Album, Playlist, Artist];
+
+  /// An icon for this content type.
+  IconData get contentIcon => ContentUtils.contentIcon(runtimeType);
```

Comment:
Song.icon, Album.icon, etc. are static constants which can be used in constant context, while `contentIcon` is a dynamic getter.

I'm not sure why would we move this into SongOrigin or PersistentQueue, since SongOrigin excludes Song and PersistentQueue exlcudes both Song and Artist.